--- x-toolchain-slk-14.1/x-toolchain/x-build	2013-10-21 08:41:49.000000000 +0100
+++ x-toolchain-slk-14.1-patched/x-toolchain/x-build	2015-04-26 00:32:16.405469518 +0100
@@ -63,7 +63,7 @@
 #BINUTILSVER=2.22.52.0.2
 #BINUTILSVER=2.23.51.0.6
 BINUTILSVER=2.23.52.0.1
-GCCVER=4.8.2
+GCCVER=4.8.4
 GLIBCVER=2.17
 
 ################## Functions ##############################################
@@ -339,14 +339,14 @@
 echo "Unpacking gcc source..."
 tar xf $CWD/sources/gcc/gcc-$GCCVER.tar.xz || exit 1
 # Add this symlink to ease application of the patches from debian:
-rm -rf src
-ln -vfs gcc-$GCCVER src 
+mv gcc-$GCCVER src
+ln -vfs src gcc-$GCCVER
 cd gcc-* || exit 1
 
 ########### Apply patches #################################################
 
 # Apply Debian patches:
-xz -dc $CWD/sources/gcc/gcc*diff.xz | patch -p1 || exit 1
+tar xvvf $CWD/sources/gcc/patches-*gcc*.xz || exit 1
 
 # Go back into the parent directory
 cd ..
@@ -354,7 +354,6 @@
 # This list is taken from the Debian Build log for the particular
 # patch release we're using, with a few redundant ones stripped.
 for patchf in \
-  svn-updates.diff \
   gcc-gfdl-build.diff \
   gcc-textdomain.diff \
   gcc-driver-extra-langs.diff \
@@ -372,31 +371,44 @@
   note-gnu-stack.diff \
   libgomp-omp_h-multilib.diff \
   sparc-force-cpu.diff \
-  pr24619.diff \
   pr47818.diff \
   pr49944.diff \
-  mudflapth-link.diff \
-  libmudflap-x32.diff \
   libgo-testsuite.diff \
   gcc-target-include-asm.diff \
   libgo-revert-timeout-exp.diff \
   arm-sanitizer.diff \
-  aarch64-libjava.diff \
   libgo-setcontext-config.diff \
   pr57211.diff \
-  gcc-auto-build.diff \
   kfreebsd-unwind.diff \
   libgcc-no-limits-h.diff \
   kfreebsd-boehm-gc.diff \
+  hurd-boehm-gc.diff \
   pr49847.diff \
   libffi-m68k.diff \
+  m68k-picflag.diff \
+  sys-auxv-header.diff \
+  go-use-gold.diff \
+  go-testsuite.diff \
+  pr57363.diff \
+  fix-ffi_call_VFP-with-no-VFP-argument.diff \
+  pr61106.diff \
+  pr61126.diff \
+  pr57653.diff \
+  libitm-aarch64.diff \
+  pr61046.diff \
+  pr61336.diff \
+  gcc-setmultilib-fix.diff \
+  pr61841.diff \
+  pr62032.diff \
+  pr59586.diff \
   libjava-armel-unwind.diff \
+  goarch-aarch64.diff \
+  libgo-explicit-reservation.diff \
   ada-kfreebsd.diff \
   arm-multilib-defaults.diff \
   gcc-ice-hack.diff \
   gcc-ice-apport.diff \
   libjava-fixed-symlinks.diff \
-  libstdc++-arm-wno-abi.diff \
   ada-mips.diff \
   libffi-ro-eh_frame_sect.diff \
   libjava-multiarch.diff \
